hpx/lcos/detail/async_implementations.hpp

```diff
@@ -41,31 +43,21 @@ namespace hpx { namespace detail
     struct sync_local_invoke
     {
         template <typename ...Ts>
-        static lcos::future<Result> call(
-            naming::id_type const& id, naming::address && addr,
-            Ts &&... vs)
+        static lcos::future<Result>
+        call(naming::id_type const& id, naming::address && addr, Ts &&... vs)
         {
-            lcos::packaged_action<Action, Result> p;
             bool target_is_managed = false;
+            naming::id_type id1;
 
             if (id.get_management_type() == naming::id_type::managed)
             {
-                naming::id_type id1(id.get_gid(), naming::id_type::unmanaged);
-                if (addr)
-                {
-                    p.apply(launch::sync, std::move(addr), id1,
-                        std::forward<Ts>(vs)...);
-                }
-                else
-                {
-                    p.apply(launch::sync, id1, std::forward<Ts>(vs)...);
-                }
+                id1 = naming::id_type(id.get_gid(), naming::id_type::unmanaged);
                 target_is_managed = true;
             }
-            else
-            {
-                p.apply(launch::sync, id, std::forward<Ts>(vs)...);
-            }
+
+            lcos::packaged_action<Action, Result> p;
+            p.apply(std::move(addr), target_is_managed ? id1 : id,
```

Comment:
Am 08.01.2016 11:19 nachm. schrieb "Hartmut Kaiser" <
notifications@github.com>:
>
> In hpx/lcos/detail/async_implementations.hpp:
>
> >                  target_is_managed = true;
> >              }
> > -            else
> > -            {
> > -                p.apply(launch::sync, id, std::forward<Ts>(vs)...);
> > -            }
> > +
> > +            lcos::packaged_action<Action, Result> p;
> > +            p.apply(std::move(addr), target_is_managed ? id1 : id,
>
> Remote operations don't support launch policies, they never had.

If I'm not mistaken, that's for the local case, no? If launch policies are
not supported, why care for deferred?
It sounds a bit contradicting...

>
> —
> Reply to this email directly or view it on GitHub.